### PR TITLE
Update 2022 02 02

### PR DESCRIPTION
**Pull request by kba**
New core v2.30.0, faster dinglehopper, new OCR alignment processor `ocrd-cor-asv-ann-align`, new features for workflow-configuration and lots of fixes in general.

### CODE REVIEW

**Comment by bertsky** [comment]
I suggest adding ocrd_detectron2 and ocrd_doxa, too.

**Comment by kba** [comment]
> I suggest adding ocrd_detectron2 and ocrd_doxa, too.

Sure, will do after the tech call.

**Comment by bertsky** [comment]
> context deadline exceeded

That's the second time in a row. Maybe should look into the total time constraints per job...

**Comment by bertsky** [comment]
> > context deadline exceeded
> 
> That's the second time in a row. Maybe should look into the total time constraints per job...

[Sounds](https://support.circleci.com/hc/en-us/articles/4410707277083-Context-deadline-exceeded-after-1-hour-Build-timed-out-Free-tier-only-) like we'd need to pay for longer than 1h per job.

Then let's simply switch to `make -j` and :crossed_fingers:

**Comment by kba** [comment]
> [Sounds](https://support.circleci.com/hc/en-us/articles/4410707277083-Context-deadline-exceeded-after-1-hour-Build-timed-out-Free-tier-only-) like we'd need to pay for longer than 1h per job.

> This is part of an update to our new pricing plans. In particular, builds on the Free plan are limited to a maximum duration of 1 hour.

Looks like we need to switch CI providers :(

File: Makefile [hunk 1]
```diff
@@ -479,6 +496,13 @@ else
 endif
 endif
 
+ifneq ($(findstring ocrd_doxa, $(OCRD_MODULES)),)
+OCRD_EXECUTABLES += $(OCRD_DOXA)
+OCRD_DOXA := $(BIN)/ocrd-doxa-binarize
+$(OCRD_DOXA): ocrd_doxa $(BIN)/ocrd
+	. $(ACTIVATE_VENV) && $(MAKE) -C $< install
```
Comment (bertsky) [inline Makefile:503]: nothin fancy here:
```suggestion
	$(pip_install)
```

**Comment by stweil** [comment]
What about updating to tesseract 5.0.1, too.

Note that there is currently a broken build (assertion) for ocrd_pc_segmentation because latest pip breaks pip-tools (see https://github.com/jazzband/pip-tools/issues/1558), so a simply `make all` stops early.

**Comment by kba** [comment]
> Note that there is currently a broken build (assertion) for ocrd_pc_segmentation because latest pip breaks pip-tools (see [jazzband/pip-tools#1558](https://github.com/jazzband/pip-tools/issues/1558)), so a simply `make all` stops early.

Thanks for the info, we also noticed this yesterday. A PR fixing this seems imminent https://github.com/jazzband/pip-tools/pull/1567

> What about updating to tesseract 5.0.1, too?

Sure, I'll add that.

**Comment by stweil** [comment]
The latest Tesseract main now includes a fix for issue https://github.com/OCR-D/ocrd_tesserocr/issues/182. @bertsky, do you think that should be included as well, or would you prefer a fix in ocrd_tesserocr?

**Comment by bertsky** [comment]
> The latest Tesseract main now includes a fix for issue [OCR-D/ocrd_tesserocr#182](https://github.com/OCR-D/ocrd_tesserocr/issues/182). @bertsky, do you think that should be included as well, or would you prefer a fix in ocrd_tesserocr?

@stweil I have not got around to test this. But since `shrink_polygons` is optional, I would not hasten this (I recall the problem I found with result iterators was quite involved)

**Comment by bertsky** [comment]
Oh, since https://github.com/tesseract-ocr/tesseract/commit/443933a75a377b608acc0e9912e792e7a97919c9 is already merged, and catching a nullptr is always a good idea – I'd say yes, lease update Tesseract once more, regardless.

File: Makefile [hunk 2]
```diff
@@ -237,6 +239,21 @@ else
 endif
 endif
 
+ifneq ($(findstring ocrd_detectron2, $(OCRD_MODULES)),)
+OCRD_EXECUTABLES += $(OCRD_DETECTRON2)
+OCRD_DETECTRON2 += $(BIN)/ocrd-detectron2-segment
+$(call multirule,$(OCRD_DETECTRON2)): ocrd_detectron2
```
Comment (stweil) [inline Makefile:245]: Is there a special reason why multirule was used here?

**Pull request merged**